cmd/util/ledger/migrations/add_missing_keys_migration.go

```diff
@@ -0,0 +1,140 @@
+package migrations
+
+import (
+	_ "embed"
+	"fmt"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-core-contracts/lib/go/templates"
+	"github.com/onflow/flow-go/fvm"
+	"github.com/onflow/flow-go/fvm/environment"
+	"github.com/onflow/flow-go/fvm/state"
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+//go:embed FlowFees.cdc
+var flowFeesContract string
+
+// AddMissingKeysAndUpgradeContractsMigration is a temporary migration that adds the missing keys to the core contract accounts
+// and upgrades the contracts to the latest version.
+// This migration is only needed for the sandboxnet but is written to also work on benchnet, so it can be tested.
+// context: https://dapperlabs.slack.com/archives/C015G65HR2P/p1671218228441159
+func AddMissingKeysAndUpgradeContractsMigration(chain flow.Chain, logger zerolog.Logger) func(payloads []ledger.Payload) ([]ledger.Payload, error) {
+	return func(payloads []ledger.Payload) ([]ledger.Payload, error) {
+		return addMissingKeysAndUpgradeContractsMigration(
+			chain,
+			payloads,
+			logger.With().Str("migration", "add-missing-keys-and-upgrade-contracts").Logger())
+	}
+}
+
+func addMissingKeysAndUpgradeContractsMigration(
+	chain flow.Chain,
+	payloads []ledger.Payload,
+	logger zerolog.Logger,
+) ([]ledger.Payload, error) {
+
+	// only run this migration on sandboxnet and benchnet
```

Comment:
this will error if you try to run migrate on other chain.  maybe change the migrate list in execution_state_extract.go instead.